keras_nlp/layers/random_swaps.py

```diff
@@ -0,0 +1,140 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+from keras import backend
+from tensorflow import keras
+
+
+class RandomSwaps(keras.layers.Layer):
+    """Augments input by randomly swapping words.
+
+    The layer works by splitting the words using `tf.strings.split` computes
+    then repeats the following n times:
+        - Chooses 2 random indices from the input.
+        - Swaps the words present at those indices.
+    These 2 randomly sampled indices can also potentially be the same index.
+
+    Args:
+        swaps: Number of swaps to perform.
+        seed: Optional random seed.
+
+
+    Examples:
+
+    Word Level usage
+    >>> inputs = tf.strings.split(["Hey I like", "Keras and Tensorflow"])
+    >>> augmenter = RandomSwaps(swaps = 3, seed = 42)
+    >>> augmented = augmenter(inputs)
+    >>> tf.strings.reduce_join(augmented, separator=" ", axis=-1)
+    <tf.Tensor: shape=(2,), dtype=string, numpy=array([b'I like Hey', b'and Tensorflow Keras'], dtype=object)>
+
+    Character Level usage
+    >>> inputs = tf.strings.unicode_split(["Hey I like", "bye bye"], "UTF-8")
+    >>> augmenter = RandomSwaps(swaps = 1, seed = 42)
+    >>> augmented = augmenter(inputs)
+    >>> tf.strings.reduce_join(augmented, axis=-1)
+    <tf.Tensor: shape=(2,), dtype=string, numpy=array([b'HeI y like', b'b eybye'], dtype=object)>
+    """
+
+    def __init__(self, swaps, seed=None, name=None, **kwargs):
+        # Check dtype and provide a default.
+        if "dtype" not in kwargs or kwargs["dtype"] is None:
+            kwargs["dtype"] = tf.int32
+        else:
+            dtype = tf.dtypes.as_dtype(kwargs["dtype"])
+            if not dtype.is_integer and dtype != tf.string:
+                raise ValueError(
+                    "Output dtype must be an integer type or a string. "
+                    f"Received: dtype={dtype}"
+                )
+
+        super().__init__(name=name, **kwargs)
+        self.swaps = swaps
+        self.seed = seed
+        self._random_generator = backend.RandomGenerator(seed)
+        if self.swaps < 0:
+            raise ValueError("Swaps must be non negative")
+
+    @tf.function
+    def call(self, inputs):
+        """Augments input by randomly swapping words.
+        Args:
+            inputs: A tensor or nested tensor of strings to augment.
+        Returns:
+            A tensor or nested tensor of augmented strings.
+        """
+
+        isString = False
+        if isinstance(inputs, str):
+            inputs = [inputs]
+            isString = True
+
+        scalar_input = inputs.shape.rank == 0
+        if scalar_input:
+            inputs = tf.expand_dims(inputs, 0)
+
+        row_splits = inputs.row_splits
```

Comment:
I think the flow of this function can look very similar to the random deletion layer up to the shuffle and trim part?